processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/MixedCaseValidatorGenerator.java

```diff
@@ -0,0 +1,92 @@
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeSpec;
+import java.util.List;
+import javax.lang.model.element.Modifier;
+import org.mobilitydata.gtfsvalidator.annotation.Generated;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.MixedCaseNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.validator.SingleEntityValidator;
+
+/**
+ * Generates validator classes to check that a string field contains mixed case characters.
+ *
+ * @see org.mobilitydata.gtfsvalidator.annotation.MixedCase
+ * @see MixedCaseNotice
+ */
+public class MixedCaseValidatorGenerator {
+  public ImmutableList<TypeSpec> generateValidator(List<GtfsFileDescriptor> fileDescriptors) {
+    ImmutableList.Builder<TypeSpec> validators = ImmutableList.builder();
+    for (GtfsFileDescriptor fileDescriptor : fileDescriptors) {
+      if (fileDescriptor.fields().stream().anyMatch(f -> f.mixedCase())) {
+        validators.add(generateValidator(fileDescriptor));
+      }
+    }
+    return validators.build();
+  }
+
+  private static TypeSpec generateValidator(GtfsFileDescriptor fileDescriptor) {
+    GtfsEntityClasses entityClasses = new GtfsEntityClasses(fileDescriptor);
+    TypeSpec.Builder typeSpec =
+        TypeSpec.classBuilder(fileDescriptor.className() + "MixedCaseValidator")
+            .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+            .addAnnotation(Generated.class)
+            .addAnnotation(GtfsValidator.class)
+            .superclass(
+                ParameterizedTypeName.get(
+                    ClassName.get(SingleEntityValidator.class),
+                    entityClasses.entityImplementationTypeName()));
+
+    MethodSpec.Builder validateMethod =
+        MethodSpec.methodBuilder("validate")
+            .addModifiers(Modifier.PUBLIC)
+            .addAnnotation(Override.class)
+            .returns(void.class)
+            .addParameter(entityClasses.entityImplementationTypeName(), "entity")
+            .addParameter(NoticeContainer.class, "noticeContainer");
+
+    for (GtfsFieldDescriptor mixedCaseField : fileDescriptor.fields()) {
+      if (!mixedCaseField.mixedCase()) {
+        continue;
+      }
+      validateMethod
+          .beginControlFlow(
+              "if (entity.$L())", FieldNameConverter.hasMethodName(mixedCaseField.name()))
+          .addStatement("$T value = entity.$L()", String.class, mixedCaseField.name())
+          .beginControlFlow("if (!(value.matches(\".*[a-z].*\") && value.matches(\".*[A-Z].*\")))")
+          .addStatement(
+              "noticeContainer.addValidationNotice(new $T(\"$L\", \"$L\", entity.csvRowNumber()))",
+              MixedCaseNotice.class,
+              fileDescriptor.filename(),
+              CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, mixedCaseField.name()))
```

Comment:
@bdferris-v2 This is what I added to get my tests to pass, but I'm not sure it's a great solution... I don't see an obvious way to validate a column header in the CSV vs. a field name in the Schema.  E.g. for `agency.txt` I don't see anywhere that the `agencyName` field is specified in code or validated that the column header is `agency_name`.  It seems it could be arbitrary. In `RowParser` we have a `CsvHeader` object where the correct header can be retrieved from for things like `MissingRequiredFieldNotice` but we don't have that in the generated validators.  

Let me know if you have any recommendations...

Reply:
I think what you is:

```
FieldNameConverter.gtfsColumnName(mixedCaseField.name())
```

It's probably a bug that this conversion is missing in `CurrencyAmountValidatorGenerator`.